### ISSUE TITLE
JIT: Consider making handles their own node type

### ISSUE THREAD

**Issue by jakobbotsch** [dotnet/runtime#104636]
The JIT has had a number of bugs related to the fact that handles are special and cannot participate in various opts, even though we represent them as `GT_CNS_INT`. #104634 is one latest example.
I think we should consider introducing a dedicated node type for the handles to make these bugs less likely to be introduced. 

cc @dotnet/jit-contrib

**Comment by SingleAccretion** [comment]
Consideration: under Jit (`!opts.compReloc`) it is ok to dissolve handles (they have their final values). It is only under AOT that handles are special.

**Comment by jakobbotsch** [comment]
@SingleAccretion Yeah, that's what I was alluding to by the "(opaque)". Though it would be nice to remove the handle flags from `GT_CNS_INT` as well, but we use those for a few things (mostly to annotate disassembly, I think), so we probably wouldn't be able to do that if we left the runtime JIT handles as `GT_CNS_INT`.
